submission/utils.py

```diff
@@ -0,0 +1,96 @@
+import glob
+import json
+import os
+import shutil
+
+
+def get_new_d3m_path():
+    """
+    Gets the name of the newest version path for d3m/byudml
+    :return:
+    """
+    new_directory = max(glob.glob('submission/primitives/v????.?.?'))
+    byu_path = "byu-dml"
+    byu_dir = os.path.join(new_directory, byu_path)
+    return byu_dir
+
+
+def create_meta_script_seed(problem):
+    """
+    Update this as time goes on.  This is only for SEED datasets
+    :param problem: the name of the problem for the dataset
+    :return: the meta file
+    """
+    return \
+        {
+        "problem": "{0}_problem".format(problem),
+        "full_inputs": [
+            "{0}_dataset".format(problem)
+        ],
+        "train_inputs": [
+            "{0}_dataset_TRAIN".format(problem)
+        ],
+        "test_inputs": [
+            "{0}_dataset_TEST".format(problem)
+        ],
+        "score_inputs": [
+            "{0}_dataset_SCORE".format(problem)
+        ]
+    }
+
+
+def extract_byu_info(pipeline_json):
+    """
+
+    :param pipeline_json: the pipeline in json form
+    :returns the python path of the BYU primitive, and the version number of the primitive
+    """
+    for primitive_step in pipeline_json["steps"]:
+        if "BYU" in primitive_step["primitive"]["python_path"]:
+            return primitive_step["primitive"]["python_path"], primitive_step["primitive"]["version"]
+
+
+def clear_directory(dir_path):
+    """
+    CAREFUL: this will DELETE ALL FILES in dirs_path
+
+    This function clears the submodule directory so that we can add the new information
+    :param dir_path: the directory where all files will be deleted
+    """
+    files = glob.glob(dir_path + '/*')
+    for f in files:
+        print("Deleting all files in", f)
+        shutil.rmtree(f)
+
+
+def create_and_add_to_directory(primitive_dir, new_version_num, pipeline_json, problem_name='185_baseball_problem'):
+    """
+    Adds pipelines to the submodule directory and creates directories if it needs it
+    :param primitive_dir: the python path of the primitive
+    :param new_version_num: the latest version number of the primitive
+    :param pipeline_json: the pipeline to be written to file
+    :param problem_name: the name of the problem
+    """
+    # make folders if they don't exist already
+    if not os.path.exists(primitive_dir):
+        os.makedirs(primitive_dir)
+
+    version_dir = os.path.join(primitive_dir, new_version_num)
+    if not os.path.exists(version_dir):
+        os.makedirs(version_dir)
+
+    pipeline_dir = os.path.join(version_dir, "pipelines")
+    if not os.path.exists(pipeline_dir):
+        os.makedirs(pipeline_dir)
```

Comment:
you don't need all these makedirs. just give it the final dir you want made and makedirs will make it recursively as needed.

Reply:
I didn't know that.  I'll make the change